cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricValue.java

```diff
@@ -124,15 +126,25 @@ public String toString() {
    * Helper function to publish the distribution metric to other systems such as Cloud Monitoring
    */
   public long[] getAllBucketCounts() {
-    return null;
+    long[] allBucketCounts = new long[NUM_FINITE_BUCKETS + 2];
+    long mask = 1;
+    int index = 0;
+    for (int i = 0; i < allBucketCounts.length; i++) {
+      if ((bucketMask & mask) >= 1) {
+        allBucketCounts[i] = bucketCounts[index];
+        index++;
+      }
+      mask = mask << 1;
+    }
+    return allBucketCounts;
   }
 
   public long getBucketMask() {
```

Comment:
I am wondering if this should be `getPositiveBucketMask` as we are not handling negative values now, but may wnt to do it later

Reply:
The mask itself is independent of bucket ranges. The bucket boundaries are sort of implicit. We may not just have positive and negative bit masks in the future. We may even have explicit boundaries. Not sure if that is the best rename.

Reply:
OK. Then add javadoc that explains it

Reply:
Added.